localstack/services/s3/provider.py

```diff
@@ -388,6 +449,89 @@ def validate_bucket_name(bucket: BucketName):
         raise ex
 
 
+def _create_invalid_argument_exc(
+    message: Union[str, None], name: str, value: str
+) -> InvalidArgument:
+    ex = InvalidArgument(message)
+    ex.ArgumentName = name
+    ex.ArgumentValue = value
+    return ex
```

Comment:
would it help if we made it easier to set the exception attributes, so you can raise
```python
raise InvalidArgument(message, ArgumentName=name, ArgumentValue=value)
```
?

there are two ways that we can do that. either we change the scaffold to create the constructors of exceptions accordingly, or we extend the `ServiceException` constructor to accept `kwargs`, to something like:

```python
    def __init__(self, *args, **kwargs):
        super(ServiceException, self).__init__(*args)

        if len(args) >= 1:
            self.message = args[0]
        else:
            self.message = ""
         
        for k,v in kwargs.items():
            setattr(self, k) = v
```

the second option is simpler and also easier to implement, however it makes it harder to subclass or customize the constructor. the first approach would maybe be cleaner but also a bit more effort.